deepchecks/vision/metrics_utils/detection_precision_recall.py

```diff
@@ -136,87 +135,94 @@ def _group_detections(self, detected, ground_truth):
         """Group gts and dts on a imageXclass basis."""
         # Calculating pairwise IoUs on classes
         bb_info = build_class_bounding_box(detected, ground_truth)
-        ious = {k: compute_pairwise_ious(**v) for k, v in bb_info.items()}
+        ious = {k: compute_pairwise_ious(v["detected"], v["ground_truth"]) for k, v in bb_info.items()}
 
         for class_id in ious.keys():
-            ev = self._evaluate_image(
+            image_evals = self._evaluate_image(
                 bb_info[class_id]["detected"],
                 bb_info[class_id]["ground_truth"],
                 ious[class_id]
             )
 
             acc = self._evals[class_id]
-            acc["scores"].append(ev["scores"])
-            acc["matched"].append(ev["matched"])
-            acc["NP"].append(ev["NP"])
+            acc["scores"].append(image_evals["scores"])
+            acc["matched"].append(image_evals["matched"])
+            acc["NP"].append(image_evals["NP"])
 
-    def _evaluate_image(self, det, gt, ious):
+    def _evaluate_image(self, detections, ground_truths, ious):
         """Det - [x, y, w, h, confidence, label], gt - [label, x, y, w, h]."""
         # Sort detections by increasing confidence
-        det = [self.Prediction(d) for d in det]
-        det_sort = np.argsort([-d.confidence for d in det], kind="stable")
+
+        detections = [self.Prediction(d) for d in detections]
+        sorted_detection_ids = np.argsort([-d.confidence for d in detections], kind="stable")
         orig_ious = ious
-        orig_gt = gt
+        orig_gt = ground_truths
+        ground_truth_area = np.array([g[3] * g[4] for g in orig_gt])
 
         scores = {}
         matched = {}
         n_gts = {}
         for min_iou in self.iou_thresholds:
-            for dets in self.max_dets:
-                for area_size in ["small", "medium", "large", "all"]:
+            for top_n_detections in self.max_detections_per_class:
```

Comment:
this was a bug, in the default case we run on only `"all"`  but we always processed every step